pytest_doctestplus/plugin.py

```diff
@@ -192,22 +203,36 @@ def parse(self, s, name=None):
             skip_next = False
             skip_all = False
 
+            comment_char = '..'
             for entry in result:
                 if isinstance(entry, six.string_types) and entry:
+                    file_format = config.getoption('text_file_format', None)
+
+                    if file_format in comment_characters:
+                        comment_char = comment_characters[file_format]
```

Comment:
It seems like this can probably happen just once outside of the loop instead of in every iteration.

Reply:
indeed. This was a remnant from the start when I still didn't see all the inner workings of this.